src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,69 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SPLIT_CHARACTER = ",";
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final int TRANSACTION_TYPE_INDEX = 0;
+    private static final int TRANSACTION_VALUE_INDEX = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String fileContent = readCsvFile(fromFileName);
+        String report = createReport(fileContent);
+        writeToFile(report,toFileName);
+    }
+
+    private String readCsvFile(String fromFileName) {
+        String line;
+        StringBuilder stringFileContent = new StringBuilder();
+
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            while ((line = bufferedReader.readLine()) != null) {
+                stringFileContent.append(line).append(System.lineSeparator());
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from the file " + fromFileName, e);
+        }
+        return stringFileContent.toString();
+    }
+
+    private String createReport(String fileContent) {
+        String[] transactions = fileContent.split(System.lineSeparator());
```

Comment:
Always try to use words from the business task. So, 'operation' is more suitable than 'transaction' here and for constants (TRANSACTION_TYPE_INDEX)
![image](https://github.com/mate-academy/jv-work-with-file/assets/20326858/4a4b942c-7ad8-473c-b311-23271d2a7c9c)